src/peft/peft_model.py

```diff
@@ -109,6 +113,7 @@ def __init__(self, model: PreTrainedModel, peft_config: PeftConfig, adapter_name
         self.peft_config = {}
         self.active_adapter = adapter_name
         self.peft_type = peft_config.peft_type
+        self.parent_class = type(model)  # Gets the parent class of the original model to update signature
```

Comment:
I think we can remove this, instead getting the parent class inside of `update_forward_signature` by calling `type(model.get_base_model())`.

Reply:
Removed and used proposed solution